packages/frontend/src/components/Explorer/ExplorePanel/index.tsx

```diff
@@ -101,11 +110,18 @@ const ExplorePanel: FC<ExplorePanelProps> = memo(({ onBack }) => {
                 replaceFields({
                     customMetrics: fieldsToReplace,
                 });
+                track({
+                    name: EventName.CUSTOM_FIELDS_REPLACEMENT_APPLIED,
```

Comment:
until you save the chart, this even tis going to keep happening, firing the event mulitple times, right ? 
If so, we should add an extra prop like the chartId so we can tell if they are duplicates

Reply:
done

Reply:
what does this useEffect do?

Reply:
We check what custom metrics have matching metrics (from yml) and then replace them. 
Check this video: https://lightdash.slack.com/archives/C08B8SMQL9L/p1739301292611069